cmd/restic/cmd_rewrite_integration_test.go

```diff
@@ -170,5 +171,133 @@ func TestRewriteSnaphotSummary(t *testing.T) {
+	rtest.OK(t, runRewrite(context.TODO(), RewriteOptions{
+		Forget:                true,
+		IncludePatternOptions: filter.IncludePatternOptions{Includes: []string{"empty-directory"}}},
+		env.gopts,
+		[]string{snapshots[0].Str()}))
+	newSnapshots := testListSnapshots(t, env.gopts, 1)
+	rtest.Assert(t, snapshots[0] != newSnapshots[0], "snapshot id should have changed")
+
+	// read restic ls output and count lines which contain the string "empty-directory"
+	out := testRunLsWithOpts(t, env.gopts, LsOptions{}, []string{newSnapshots[0].String()})
+	fileList := strings.Split(string(out), "\n")
+	count := 0
+	for _, filename := range fileList {
+		if strings.Contains(filename, "empty-directory") {
+			count++
+		}
+	}
+	rtest.Assert(t, count == 1, "there should be 1 empty directory in the snapshot, but there are %d snapshots", count)
+
+	// get snapshot summary and find this 1 directory
+	_, repo, unlock, err := openWithExclusiveLock(context.TODO(), env.gopts, false)
+	rtest.OK(t, err)
+
+	sn, err := restic.LoadSnapshot(context.TODO(), repo, newSnapshots[0])
+	rtest.OK(t, err)
+
+	rtest.Assert(t, sn.Summary != nil, "snapshot should have a summary attached")
+	rtest.Assert(t, sn.Summary.TotalFilesProcessed == 0,
+		"there should be 0 files in the snapshot, but there are %d files", sn.Summary.TotalFilesProcessed)
+	unlock()
+}
+
+func TestRewriteConflictingOptions(t *testing.T) {
+	env, cleanup := withTestEnvironment(t)
+	defer cleanup()
+	createBasicRewriteRepo(t, env)
+
+	err := runRewrite(context.TODO(), RewriteOptions{}, env.gopts, []string{"latest"})
+	rtest.Assert(t, err != nil, "Nothing to do: no includes/excludes provided and no new metadata provided")
+
+	err = runRewrite(context.TODO(), RewriteOptions{
+		IncludePatternOptions: filter.IncludePatternOptions{Includes: []string{"JohannSebastianBach"}},
+		ExcludePatternOptions: filter.ExcludePatternOptions{Excludes: []string{"WolfgangAmadeusMozart"}},
+	}, env.gopts, []string{"latest"})
+	rtest.Assert(t, err != nil, "You cannot specify include and exclude options simultaneously!")
+
+	err = runRewrite(context.TODO(), RewriteOptions{
+		SnapshotSummary:       true,
+		IncludePatternOptions: filter.IncludePatternOptions{Includes: []string{"JohannSebastianBach"}},
+	}, env.gopts, []string{"latest"})
+	rtest.Assert(t, err != nil, "You cannot specify include or exclude options together with --snapshot-summary!")
+}
+
+func TestRewriteIncludeNothing(t *testing.T) {
+	env, cleanup := withTestEnvironment(t)
+	defer cleanup()
+	createBasicRewriteRepo(t, env)
+	testListSnapshots(t, env.gopts, 1)
+
+	// restic rewrite latest -i nothing-whatsoever --forget
+	rtest.OK(t, runRewrite(context.TODO(), RewriteOptions{
+		Forget:                true,
+		IncludePatternOptions: filter.IncludePatternOptions{Includes: []string{"nothing-whatsoever"}}},
+		env.gopts,
+		[]string{"latest"}))
+	testListSnapshots(t, env.gopts, 1)
+}
+
+func TestRewriteExcludeNothing(t *testing.T) {
```

Comment:
I'm not sure what this test case actually tests. Even `--exclude *` would not remove a snapshot as `KeepEmptyDirectories` is always true for excludes. Or is that what you wanted to check? Then the exclude pattern must be `*`.